stock_storage_type_putaway_abc_classification_sale_stock/models/stock_location.py

```diff
@@ -0,0 +1,24 @@
+# Copyright 2022 ACSONE SA/NV
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+from odoo import models
+
+
+class StockLocation(models.Model):
+
+    _inherit = "stock.location"
+
+    def _get_sorted_leaf_locations_orderby_abc(self, products):
+        """
+        'self' is the location we want to put products and apply the
+        putaway strategy.
+        Evaluate the abc classification profile ids in order to retrieve
+        the corresponding one.
+        """
+        location_level = products.abc_classification_product_level_ids.filtered(
```

Comment:
@rousseldenis product**s** means potentially a recordset with more than one product... Do we have to enforce that the method is always called with 1 product our do we have to take the first product into the recordset as into the initial implementation

Reply:
@jbaudoux Your input on this ? IMHO, we indeed could use ensure_one() as we must ensure location_level is 1.